components/stepline.tsx

```diff
@@ -0,0 +1,152 @@
+import { cn } from "./../lib/utils";
+import {
+  Children,
+  cloneElement,
+  Fragment,
+  isValidElement,
+  ReactNode,
+  useState,
+} from "react";
+
+export interface SteplineProps {
+  children?: ReactNode;
+  className?: string;
+  reversable?: boolean;
+}
+
+export interface SteplineItemProps {
+  title?: string;
+  subtitle?: ReactNode[];
+  variant?: "current" | "todo" | "error" | "completed";
+  className?: string;
+  onClick?: () => void;
+  id?: number;
+}
+
+function Stepline({ children, className, reversable }: SteplineProps) {
+  const [hoveredIndex, setHoveredIndex] = useState<number | null>(null);
+
+  const childArray = Children.toArray(children).filter(isValidElement);
+
+  return (
+    <div
+      className={cn(
+        "flex flex-row gap-2 py-2 px-2 relative overflow-x-auto scrollbar-thin-x",
+        className
+      )}
+    >
+      {childArray.map((child, index) => {
+        if (
+          !isValidElement<
+            SteplineItemProps &
+              Partial<{
+                hoveredIndex?: number | null;
+              }>
+          >(child)
+        )
+          return null;
+        const variant = child.props.variant;
+        const onClick = child.props.onClick;
+
+        return (
+          <div
+            key={index}
+            onClick={() => {
+              if (reversable && variant) {
+                onClick();
+              }
+            }}
+            onMouseEnter={() => {
+              setHoveredIndex(index + 1);
+            }}
+            onMouseLeave={() => {
+              setHoveredIndex(null);
+            }}
+            className={cn(
+              "flex flex-row gap-2 relative",
+              reversable && variant && "cursor-pointer"
+            )}
+          >
+            <div className="flex flex-row w-full justify-between">
+              {cloneElement(child, {
+                id: index + 1,
+                hoveredIndex: hoveredIndex,
+              })}
+            </div>
+            <div className="flex flex-row relative items-center w-full">
+              {index !== childArray.length - 1 && (
+                <div
+                  className={cn(
+                    "flex-1 h-px min-w-[44px] bg-gray-400",
+                    variant === "error" && "bg-[#b60000]",
+                    (variant === "completed" || variant === "current") &&
```

Comment:
is it possible to have function like `getBackgroundColorByVariant` and `getInnerCircleBackgroundColorByVariant` and use those functions on stepline and timeline to ensure coloring consistency.

if it's possible doing so, but by not using `bg-[]` style of setting, I don't mind specifying the coloring as `style={{}}` kind of setting.